### ISSUE TITLE
CI fails on Canberra metric tests on XLA

### ISSUE THREAD

**Issue by vfdev-5** [pytorch/ignite#1696]
## 🐛 Bug description

Canberra metric tests on XLA are failing:
```
>       assert canberra.pairwise([np_y_pred, np_y])[0][1] == pytest.approx(res)
E       assert 4.911688311688311 == nan ± ???
E         +4.911688311688311
E         -nan ± ???
```
https://github.com/pytorch/ignite/pull/1690/checks?check_run_id=1966203329

cc @sdesrozis

**Comment by vfdev-5** [comment]
Probably, the issue is related to the pull request where it happens.

**Comment by sdesrozis** [comment]
It seems weird because the mentioned PR not related to Canberra...

**Comment by vfdev-5** [comment]
Seems like the issue is here:
```python
errors = torch.abs(y - y_pred) / (torch.abs(y_pred) + torch.abs(y))
```
where the random data gives `x / tensor([ 4.,  5., 11.,  7.,  0., 10., 10.,  8.,  7.,  7.], device='xla:0')` -> nan

**Comment by KickItLikeShika** [comment]
@sdesrozis, Yes it's definitely weird! i have not touched the Canberra! And now i'm working on other metrics #1695, i'm excited what's going to happen when i finish the tests and send the PR tomorrow, probably this we more things more clear, it would show us where is the problem exactly

**Comment by vfdev-5** [comment]
Fixed by #1699
